autometrics-macros/src/lib.rs

```diff
@@ -40,22 +41,22 @@ pub fn autometrics(
     output.into()
 }
 
-fn check_async_trait(input: proc_macro::TokenStream) -> (bool, proc_macro::TokenStream) {
-    let str = input.to_string();
-
-    if str.contains("#[async_trait]") || str.contains("#[async_trait::async_trait]") {
-        // .unwrap is safe because we only remove tokens from the existing stream, we dont add new ones
-        (
-            true,
-            proc_macro::TokenStream::from_str(
-                &str.replace("#[async_trait]", "")
-                    .replace("#[async_trait::async_trait]", ""),
-            )
-            .unwrap(),
-        )
-    } else {
-        (false, input)
-    }
+/// returns a tuple of two containing:
+/// - `async_trait` attributes that have to be re-added after our instrumentation magic has been added
+/// - `input` but without the `async_trait` attributes
+fn check_async_trait(input: proc_macro::TokenStream) -> (String, proc_macro::TokenStream) {
+    // .unwrap is safe because the regex is hardcoded and thus guaranteed to be successfully parseable
+    let regex = Regex::new(r#"#\[[^\]]*async_trait\]"#).unwrap();
```

Comment:
When you have a comment about "why an `unwrap` is safe", I'd rather use an `expect` with the explanation in the message. This way, if we hit an odd case (or get "code desync"), the panic message will directly say what's wrong in the logs, and it's easier to search for the location in the codebase using the panic message as a starter
```suggestion
    let regex = Regex::new(r#"#\[[^\]]*async_trait\]"#).expect("The regex is hardcoded and thus guaranteed to be successfully parseable");
```